### DISCUSSION TITLE
When Q-dialog is used in iFrame - dialog exceeds viewport

### DISCUSSION THREAD

**Discussion by bitmandk** [quasarframework/quasar#15250] [General - Components / Directives / etc]
I have an existing solution were I gradually replaces existing html parts with Quasar components. The CSS in the existing solution does not work with Quasar, so I have to "iFrame" all my Quasar-parts.

When using q-dialog together with full-size, full-height the q-dialog get's way bigger than visible area on the screen, when "iFramed"

[Without iFrame](https://demo1.botjek.dk/test/framecontent.html)
In the above link, Q-dialog functionality in a plain htlm page - works as expected (q-dialog takes up the available space) - click "Show dialog" to see the dialog.

But if I take the plain HTML page from above (https://demo1.botjek.dk/test/framecontent.html) and places it in a iFrame (as shown below) the q-dialog gets way bigger thant the visible area on screen.

[When used in iFrame](https://demo1.botjek.dk/test/frameparent.html)

`<div id="parent" style="height:2300px;">
        <iframe style="height: 100%; width: 100%;" height="100%" width="100%" id="frame" src="framecontent.html" frameborder="0" scrolling="no"></iframe>
    </div>`

I have tried a lot of styling hacks to get the q-dialog to fit the page, but nothing seems to fix it, is it even possible?

**Comment by pdanpdan** [comment]
The size of the iframe is the size of the window for what is inside iframe
You must adjust the size of the iframe to fit on screen

**Reply by bitmandk** [reply]
@pdanpdan Waht do you mean by "adjust the size of the iframe to fit on screen"? The iframe will always be > 2000px as the page contains a list of items.

**Comment by bitmandk** [comment]
The strange thing is, that [Without iFrame](https://demo1.botjek.dk/test/framecontent.html) and [When used in iFrame](https://demo1.botjek.dk/test/frameparent.html) has identical implementations. Height of content are identical on the 2 pages only exceptions is that one page is iFramed and that changes the behavior of the q-dialog.

Can anybody explain why?

**Comment by pdanpdan** [comment]
As I already mentioned, the iframe is the window for its content
You have `<iframe style="height: 100%; width: 100%;" height="100%" width="100%" id="frame" src="framecontent.html" frameborder="0" scrolling="no"></iframe>` and that setting for height fills the whole parent's height
Use something like `<iframe style="height: 100vh; width: 100%;" height="100%" width="100%" id="frame" src="framecontent.html" frameborder="0" scrolling="no"></iframe>` if lazy or determine the available height and update iframe's height
